src/main/java/seedu/address/logic/commands/WorkerAvailableCommand.java

```diff
@@ -0,0 +1,164 @@
+     */
+    public WorkerAvailableCommand(Index targetIndex, Role role) {
+        this.targetIndex = targetIndex;
+        this.role = role;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Shift> lastShownShiftList = model.getFilteredShiftList();
+
+        if (targetIndex.getZeroBased() >= lastShownShiftList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_SHIFT_DISPLAYED_INDEX);
+        }
+
+        if (!model.hasRole(this.role)) {
+            throw new CommandException(String.format(Messages.MESSAGE_ROLE_NOT_FOUND, role));
+        }
+
+        Shift selectedShift = lastShownShiftList.get(targetIndex.getZeroBased());
+        if (!selectedShift.isRoleRequired(this.role)) {
+            throw new CommandException(MESSAGE_ROLE_FULL_OR_NOT_REQUIRED);
+        }
+
+        List<Pair<Worker, Index>> availableWorkers = findAvailableWorkers(model, selectedShift);
+
+        if (availableWorkers.size() == 0) {
+            return new CommandResult(
+                    String.format(MESSAGE_NO_AVAIL_WORKERS_SUCCESS,
+                            this.targetIndex.getOneBased(), this.role.getRole()));
+        }
+
+        String printableListOfAvailableWorkers = printListOfAvailableWorkers(availableWorkers);
+
+        return new CommandResult(
+                String.format(MESSAGE_HAS_AVAIL_WORKERS_SUCCESS, this.targetIndex.getOneBased(),
+                        this.role.getRole(), printableListOfAvailableWorkers));
+    }
+
+    private List<Pair<Worker, Index>> findAvailableWorkers(Model model, Shift selectedShift) {
+        List<Pair<Worker, Index>> availableWorkers = new ArrayList<>();
+
+        List<Worker> lastShownWorkerList = model.getFilteredWorkerList();
+        List<Assignment> fullAssignmentList = model.getFullAssignmentList();
+
+        List<Assignment> assignmentsForSelectedShift = new ArrayList<>();
+
+        for (Assignment assignment : fullAssignmentList) {
+            if (selectedShift.equals(assignment.getShift())) {
+                assignmentsForSelectedShift.add(assignment);
+            }
+        }
+
+        for (int i = 0; i < lastShownWorkerList.size(); i++) {
+            Worker worker = lastShownWorkerList.get(i);
+            if (checkIfWorkerAlreadyAssigned(assignmentsForSelectedShift, worker) || !checkIfWorkerHasRole(worker)) {
+                continue;
+            }
+
+            Set<Unavailability> workerUnavailableTimings = worker.getUnavailableTimings();
+
+            boolean isWorkerAvailable = true;
+
+            for (Unavailability unavail : workerUnavailableTimings) {
+                if (selectedShift.getShiftDay().equals(unavail.getDay())
+                        && selectedShift.getShiftTime().equals(unavail.getTime())) {
+                    isWorkerAvailable = false;
+                    break;
+                }
+            }
```

Comment:
You can also use Worker#isFitForRole() and Worker#isUnavailable() to check for role and leave availabilities